decode.go

```diff
@@ -142,14 +147,14 @@ func readToWithErrorHandler(decoder Decoder, errHandler ErrorHandler, out interf
 		return err
 	}
 	if len(csvRows) == 0 {
-		return errors.New("empty csv file given")
+		return ErrEmptyCSVFile
 	}
 	if err := ensureOutCapacity(&outValue, len(csvRows)); err != nil { // Ensure the container is big enough to hold the CSV content
 		return err
 	}
 	outInnerStructInfo := getStructInfo(outInnerType) // Get the inner struct info to get CSV annotations
 	if len(outInnerStructInfo.Fields) == 0 {
-		return errors.New("no csv struct tags found")
+		return ErrEmptyCSVFile
```

Comment:
Had to parse your library to find that there is the wrong error here ! must be ErrNoStructTags not ErrEmptyCSVFile